### ISSUE TITLE
Dark mode sometimes is applied in disable website after open them in a new tab [BUG]

### ISSUE THREAD

**Issue by DanielRios549** [darkreader/darkreader#3713]
### **Describe the bug**

Acting on sites in blacklist when opening them in a new tab using mouse scroll button.

### **Steps to reproduce**

- Add a random website to the blacklist.
- Click this site and open some internal link using the mouse scroll button to open it in a new tab.

### **Expected behavior**

The plugin should do not interfere in websites on the blacklist.

### **Actual behavior**

The extension applies the style to the page just open in a new tab, after you reload or deactivate and activate the plugin again, it stops.

### **System Information:**
 - OS: KDE Neon 5.19
 - Browser: Firefox 80 Flatpak
 - Dark Reader version: 4.9.20

### **Additional context**

This bug does not occur all the time, when I was writing this bug report it does not occurs again to make a screenshot, but 5 minutes before it does. And it also occurs using Windows 10 and others browsers, but as it is now, it is only sometimes.

**Comment by TheTural** [comment]
Confirming I am also seeing this issue.

I can reliably replicate the problem with the following behavior in Firefox 81.0 on Windows 10 (DR version 4.9.20):

1. Open a site.
2. Middle click on a link to open the page in a background tab
    I assume it would not work if you have a setting enabled to auto-switch to new tabs
3. Let the page completely load
4. Switch to the tab
5. Observe that Dark Reader is applying a dark background to the page

I have found that if I switch to the tab before it's fully loaded, the bug does not occur. I am not having the issue when a page is the foreground tab when it finishes loading, which is why refreshing the page seems like a fix.

Screenshot of the bug:
![Bug](https://user-images.githubusercontent.com/18756243/94326059-57531b00-ff67-11ea-963c-d46116b9e789.png)

Expected/correct page load:
![Correct](https://user-images.githubusercontent.com/18756243/94326076-6e920880-ff67-11ea-99ab-2afcafe29ae4.png)

**Comment by sandorex** [comment]
I have also experienced this, i had to disable dark reader
Although in my case it usually happens on youtube

**Comment by BrianPeek** [comment]
+1, haven't found a repro, but has been occurring on 2 machines for me most of today...

**Comment by brizaldi** [comment]
yup, can confirm, using Firefox 81.0 on Windows 10, it only happen sometimes when i open link in a new tab

**Comment by aspenluxxxy** [comment]
Firefox 81.0 on Arch Linux, happening to me too.

Just started today seemingly.

**Comment by VFDan** [comment]
Can confirm with Firefox 81.0 on 64-bit Windows 10, it even happens when I turn the whole extension off.

**Comment by VFDan** [comment]
Just saying, it breaks whether I switch to the v5beta or not.

**Comment by eldarshamukhamedov** [comment]
Confirmed on Windows 10 and macOS Catalina, running FF Dev version 82.0b3 (64-bit).

**Comment by Gusted** [comment]
Hi!

I've opened up a PR(#3715) which might fix this.
If someone could be so nice and try to side-load the fix by:

[build-firefox.zip](https://github.com/darkreader/darkreader/files/5286111/build-firefox.zip)

- Unpack build-firefox.zip into a folder and remember where this folder is located.

- Open the hamburger menu in the right upper corner. ![](https://i.imgur.com/ALb9GMU.png)
- Click on Addons. ![](https://i.imgur.com/565La4k.png)
- Click on the setting icon. ![](https://i.imgur.com/dA0XZ3G.png)
- Click on `Debug Add-ons`. ![](https://i.imgur.com/6AmFnyY.png) **WARNING SOMEHOW THIS IS A WHITE-ISH SITE**
- Click on `Load Temporary Add-On`. ![](https://i.imgur.com/J7PCwsw.png)
- Now go to the unpacked folder and select `manifest.json`. ![](https://i.imgur.com/WrloLPR.png)
- Click on `Open`.
- Poof and side-loaded the extension congratulations.

Regards,
Gusted

**Comment by CorePoint** [comment]
That seems to have solved it. :)

**Comment by Xeteon** [comment]
Same issue on Firefox v81.0 / Windows 10 / Dark Reader v4.9.20.

**Comment by Yash-Singh1** [comment]
For me, the issue runs on only some websites.

**Comment by alexanderby** [comment]
The version 4.9.21 with a fix is available in the store.

**Comment by DanielRios549** [comment]
Wow, you were fast, less than 24 hours and the fix is ready, thanks too much for that.

**Comment by Gusted** [comment]
> Wow, you were fast, less than 24 hours and the fix is ready, thanks too much for that.

Well, we had this bug once before(The fallback logic) with #3575 #3577 that it randomly was dark, without any reason what so ever even if the whole extension was disabled.
And after [this](https://github.com/darkreader/darkreader/issues/3713#issuecomment-699247797) that posted a screenshot I clearly could see the fallback script was being injected(as it's very minimalism to prevent white-flashes). So I guessed it was not being removed(but as I hadn't any reproduce-case as it seems to be kind of a race condition error). It probably had to due with the new Mutation Observer that was implemented with #3707. That was injecting the script anyway without checking if it was already removed(clean-upped by the other content script when the extension is disabled or site doesn't have to be darken). 

So after woken up from 10 mail notifications of this issue(Never forget to put your phone on silent people!) around 4 am and realizing that ^, I made a fix(#3715) to check if the element is still in the DOM(and not removed).

So after that, I start worrying @alexanderby didn't saw the PR(Weekend is weekend 😄) he was somehow online 2 hours after I've made the #3715. He could get update 4.9.21 out and all this just under 24 hours 😄 

Oh yeah for all optimistic people out there, we have now got quite a few breaking changes and are looking into more testes that cover real-world testing e.g. visiting certain sites. As we are currently just testing certain functions on its output, but not how it's performing as an extension in a browser. As this could easily be fixed if their was a test in place that's currently being done manual. 

I think, after all, experience and good timing made this quick fix happen. Credits to @alexanderby and to the person that reviewed our extension as that could taken day(as once again weekend is weekend).

Regards,
Gusted
